ufo2fontir/src/source.rs

```diff
@@ -614,14 +614,25 @@ impl Work<Context, WorkError> for GlobalMetricsWork {
     fn exec(&self, context: &Context) -> Result<(), WorkError> {
         debug!("Global metrics for {:#?}", self.designspace_file);
         let static_metadata = context.get_init_static_metadata();
-        let mut metrics = GlobalMetrics::new(
-            static_metadata.default_location().clone(),
-            static_metadata.units_per_em,
-        );
 
         let designspace_dir = self.designspace_file.parent().unwrap();
         let font_infos = font_infos(designspace_dir, &self.designspace)?;
         let master_locations = master_locations(&static_metadata.axes, &self.designspace.sources);
+        let Some((_, default_master)) = default_master(&self.designspace) else {
+            return Err(WorkError::NoDefaultMaster(self.designspace_file.clone()));
+        };
+
+        let mut metrics = GlobalMetrics::new(
+            static_metadata.default_location().clone(),
+            static_metadata.units_per_em,
+            font_infos
+                .get(&default_master.filename)
+                .ok_or_else(|| {
+                    WorkError::FileExpected(designspace_dir.join(&default_master.filename))
+                })?
+                .x_height
+                .map(|v| v as f32),
+        );
```

Comment:
you should add a TODO somewhere here for adding support to reading OS/2 typo/win metrics from UFO fontinfo.plist.

Maybe we need an issue in which we track the differences in the support of various features between .glyphs vs .designspace input. Since lately you're been advancing the .glyphs side of things (e.g. #293) there's a risk that the UFO+DS one lags behind.

Reply:
Will add todo.